### PR TITLE
Documentation: Added "Drafts" folder and "Documentation Guidelines.md"

### PR DESCRIPTION
**Pull request by dunkOnIT**
Implicit in some of what I wrote in this PR are a number of proposals which I believe warrant input from @lgarron and @rokicki. For the sake of expediency, I wrote them as declarative statements in Documentation Guidelines, but everything here represents a proposal for discussion.

To start, I suggest we start distinguishing between 2 kinds of documentation:
- **API Reference** - describes the usage of the functions/modules which are exposed through the `cubing.js` library
- **Long-form documentation** - README, tutorials, CONTRIBUTING.md - anything that looks more like prose than function definitions.
When I use the general term "documentation", I am referring to _both_ of these.

Overall, these proposals aim to do 3 things:
1. Present a roadmap for addressing the current documentation backlog 
2. Prevent the growth of the documentation backlog by encouraging API reference documentation "as we go"
3. Introduce the concept of "Draft" documentation to make it easy to "jot down" long-form documentation as it comes to us (as this document did to me), rather than feel that if we are going to write any documentation at all it has to be perfect.

## Proposals:
(a) All **new** modules/functions added should only be accepted/considered complete if they are accompanied by API Reference documentation - in line with (2) above
(b) Store draft documentation in `/docs/draft`, so that the drafts people come up with are shared and can be taken forward by someone else if needed
(c) Not backtick technical references when they include links, to prevent ambiguity (this comes out of my experience of trying to browse the README
(d) See `Documentation roadmap` section for my proposed roadmap for tackling the current documentation backlog

Discussion welcome, hence the draft nature of this PR. Once/if considered by both of you and agreed, we can merge.

### CODE REVIEW

File: docs/drafts/Documentation Guidelines.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+*NB: This information is still in draft form, and should not be relied upon to inform usage of or contribution to `cubing.js`.*
+
+## Purpose of this document
+
+This document serves to describe `cubing.js`'s approach to documentation in both practical and theoretical terms. Anyone looking to contribute either documentation or new features to `cubing.js` should consult this document.
+
+## Documentation guidelines:
+
+- Write API reference documentation inline using [TSDoc-formatted](https://tsdoc.org/) comments 
+    - PR's lacking these comments will be considered incomplete
```
Comment (lgarron) [inline docs/drafts/Documentation Guidelines.md:10]: I think this phrasing is a tad aggressive. And if we get a PR that we think needs documentation, a reviewer will almost certainly ask. Could we go for something like "Please include documentation for any new definitions (e.g. classes, functions, types, `const` exports) that someone using the library may interact with."

File: docs/drafts/Documentation Guidelines.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+- Decide on documentation process and tooling
+- Create library architecture overview, and create API reference for through backlog of undocumented modules/features (serves needs of experienced developers and contributors)
+- Create tutorials/guides for how to apply the library in practice (serves needs of novice developers)
+
+## Style Guidelines
+
+- Technical references (ie, references to files, projects and tools) should be formatted as follows:
+  - Use backticks \` when there is no accompanying link (eg, "`cubing.js` is a project which aims to...")
+  - Omit backticks when there is an accompanying link (eg, "Make sure you have [node](https://nodejs.org/en/) installed before...")
+  - Reasoning: Universal use of backticks disguises links, leading to confusing navigation
```
Comment (lgarron) [inline docs/drafts/Documentation Guidelines.md:28]: I've used backticks inside link text for a lot of https://github.com/cubing . I'm open to change, but I'd prefer to keep things consistent unless there's a really good reason to change. Do you have a reference to the issues this can cause?

File: docs/drafts/Documentation Guidelines.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+- Draft documentation should include the following disclaimer at the top of the document:
+  - *NB: This information is still in draft form, and should not be relied upon to inform usage of or contribution to `cubing.js`.*
```
Comment (lgarron) [inline docs/drafts/Documentation Guidelines.md:30]: Nit: Do you think you could rephrase this to use an active voice (i.e. avoid "should not be relied upon")? This would end up in a lot of places, and I think we should make it as clear as possible to non-native English speakers.

```suggestion
  - *NB: This information is still in draft form, and should not be relied upon to inform usage of or contribution to `cubing.js`.*
```

**Review by lgarron — COMMENTED** [review]
Could I ask you to move this to `/docs/README.md`?

File: docs/drafts/Documentation Guidelines.md (hunk 1 above)
Comment (TydVirTaal) [reply docs/drafts/Documentation Guidelines.md:10]: Question here - do you not think documentation should be written even on internal changes? My thinking is that this will make it easier for contributors to onboard, but people contributing to the core codebase may not necessarily be a huge priority, as you and Tomas are getting quite a bit done - and trying to build internal documentation may slow that down for relatively little gain.

Comment (TydVirTaal) [reply docs/drafts/Documentation Guidelines.md:10]: Also, see updated phrasing

File: docs/drafts/Documentation Guidelines.md (hunk 2 above)
Comment (TydVirTaal) [reply docs/drafts/Documentation Guidelines.md:28]: I've realised that the problem I experienced doesn't happen in Github itself, only in my VSCode Markdown previewer.

Github indicates backticks with more than just text highlighting, so the same issue doesn't apply. For example, this text shows all 3 formatting combinations clearly:
image

However, in VSCode, it only uses text highlighting, making the link on `src/sites/alpha.twizzle.net` ambiguous. (Obviously one can infer from context with it being a URL, but even then it could be plaintext and not a hyperlink):
image

Based on this, I think we can ignore this suggestion.

File: docs/drafts/Documentation Guidelines.md (hunk 3 above)
Comment (TydVirTaal) [reply docs/drafts/Documentation Guidelines.md:30]: Great point!
